edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/requests/LogsUploadRequestHandler.cs

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Agent.Core.Requests
+{
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Edge.Agent.Core.Logs;
+    using Microsoft.Azure.Devices.Edge.Util;
+
+    public class LogsUploadRequestHandler : RequestHandlerBase<LogsUploadRequest, object>
+    {
+        readonly ILogsUploader logsUploader;
+        readonly ILogsProvider logsProvider;
+
+        public LogsUploadRequestHandler(ILogsUploader logsUploader, ILogsProvider logsProvider)
+        {
+            this.logsProvider = Preconditions.CheckNotNull(logsProvider, nameof(logsProvider));
+            this.logsUploader = Preconditions.CheckNotNull(logsUploader, nameof(logsUploader));
+        }
+
+        public override string RequestName => "UploadLogs";
+
+        protected override async Task<Option<object>> HandleRequestInternal(Option<LogsUploadRequest> payloadOption)
+        {
+            LogsUploadRequest payload = payloadOption.Expect(() => new ArgumentException("Request payload not found"));
+            var moduleLogOptions = new ModuleLogOptions(payload.Id, payload.Encoding, payload.ContentType);
+            byte[] logBytes = await this.logsProvider.GetLogs(moduleLogOptions, CancellationToken.None);
```

Comment:
Should this do some sort of chunking/streaming, instead of loading all the logs into memory before uploading to the cloud?

Reply:
I thought about it, and we can do that. The reason I didn't do it was for compression support.. the compression is better when the entire file is compressed

Reply:
I am adding chunking support in my next PR (for log streams). I will check with Venkat and add support for chunking for blobs as well, if he feels it is useful

Reply:
👍